### PR TITLE
chore(cli): add deprecation message for introspect command

### LINKED ISSUE #8078: CLI: Add deprecation notice to prisma introspect (renamed db pull)
prisma introspect is being replaced with prisma db pull.
We should add a deprecation notice so that users are aware that the command will eventually go away in the future (not set date for sunsetting yet, we'll provide enough time).

### PR DESCRIPTION
**Pull request by Jolg42**
Screen Shot 2021-08-16 at 14 50 28

Renamed to `prisma db pull`

Closes https://github.com/prisma/prisma/issues/8078

- added test for the warning
- removed the really not nice test.sh and converted tests to Jest
- moved some tests to a commands folder like `packages/cli/src/__tests__/commands/Studio.test.ts`

### CODE REVIEW

File: packages/cli/src/Studio.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ ${chalk.bold('Examples')}
         resolve: {
           '@prisma/client': path.resolve(
             __dirname,
-            '../prisma-client/index.js',
+            '../../prisma-client/index.js',
```
Comment (Jolg42) [inline packages/cli/src/Studio.ts:123]:
```suggestion
            '../prisma-client/index.js',
```

File: packages/cli/src/Studio.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ ${chalk.bold('Examples')}
       args['--port'] || (await getPort({ port: getPort.makeRange(5555, 5600) }))
     const browser = args['--browser'] || process.env.BROWSER
 
-    const staticAssetDir = path.resolve(__dirname, '../build/public')
+    const staticAssetDir = path.resolve(__dirname, '../../build/public')
```
Comment (Jolg42) [inline packages/cli/src/Studio.ts:112]:
```suggestion
    const staticAssetDir = path.resolve(__dirname, '../build/public')
```

File: packages/cli/src/__tests__/commands/Generate.test.ts [hunk 3]
```diff
@@ -95,13 +93,13 @@ describe('--schema from parent directory', () => {
...
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-            ✔ Generated Prisma Client (0.0.0) to ./subdirectory/@prisma/client in 147ms
-            You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client
-            \`\`\`
-            import { PrismaClient } from './subdirectory/@prisma/client'
-            const prisma = new PrismaClient()
-            \`\`\`
-          `)
+✔ Generated Prisma Client (0.0.0) to ./subdirectory/@prisma/client in XXms
```
Comment (millsp) [inline packages/cli/src/__tests__/commands/Generate.test.ts:96]: I'm not sure why the the indentation changed

File: packages/cli/src/__tests__/fixtures/custom-generator/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,34 @@
+generator min {
+  // https://github.com/timsuchanek/minimal-generator
+  provider = "npx @timsuchanek/minimal-generator"
```
Comment (millsp) [inline packages/cli/src/__tests__/fixtures/custom-generator/prisma/schema.prisma]: How does this work?

File: packages/cli/src/__tests__/commands/Generate.test.ts (hunk 3 above)
Comment (Jolg42) [reply packages/cli/src/__tests__/commands/Generate.test.ts:96]: One mysterious thing for me with Jest ...

File: packages/cli/src/__tests__/fixtures/custom-generator/prisma/schema.prisma (hunk 4 above)
Comment (Jolg42) [reply packages/cli/src/__tests__/fixtures/custom-generator/prisma/schema.prisma]: It runs the command, it's what powers the go client like https://github.com/prisma/prisma-client-go/blob/master/docs/quickstart.md
```
generator db {
    provider = "go run github.com/prisma/prisma-client-go"
}
```

Comment (Jolg42) [reply packages/cli/src/__tests__/fixtures/custom-generator/prisma/schema.prisma]: Source code should be here https://github.com/prisma/prisma/blob/f731467b8ac7169bfc97212abb537f4bbf9ebc92/packages/generator-helper/src/GeneratorProcess.ts#L48-L66

File: packages/cli/src/CLI.ts [hunk 5]
```diff
@@ -48,18 +49,34 @@ export class CLI implements Command {
...
+      logger.warn('')
+      logger.warn(
+        `${chalk.bold(
+          `${chalk.underline(
+            'prisma introspect',
+          )} command is deprecated. It has been renamed to ${chalk.green(
+            'prisma db pull',
+          )}`,
+        )}`,
+      )
```
Comment (Jolg42) [inline packages/cli/src/CLI.ts:76]:
```suggestion
      logger.warn(
        `${chalk.bold(
          `The ${chalk.underline(
            'prisma introspect',
          )} command is deprecated. Please use ${chalk.green(
            'prisma db pull',
          )} instead.`,
        )}`,
      )
```

File: packages/cli/src/__tests__/commands/CLI.test.ts [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+  await expect(result).rejects.toMatchInlineSnapshot(`
+          Could not find a schema.prisma file that is required for this command.
+          You can either provide it with --schema, set it as \`prisma.schema\` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
+        `)
+  expect(ctx.mocked['console.log'].mock.calls).toHaveLength(0)
+  expect(ctx.mocked['console.info'].mock.calls).toHaveLength(0)
+  expect(ctx.mocked['console.warn'].mock.calls.join('\n'))
+    .toMatchInlineSnapshot(`
+    prisma:warn 
+    prisma:warn prisma introspect command is deprecated. It has been renamed to prisma db pull
```
Comment (Jolg42) [inline packages/cli/src/__tests__/commands/CLI.test.ts:111]:
```suggestion
    prisma:warn The introspect command is deprecated. Please use db pull instead.
```

File: packages/cli/fixtures/test.sh [hunk 7]
```diff
@@ -1,134 +0,0 @@
-#!/bin/bash
```
Comment (janpio) [inline packages/cli/fixtures/test.sh:1]: Sure this file should go away? Did these tests move somewhere else?

Comment (millsp) [reply packages/cli/fixtures/test.sh:1]: They've been ported to `src/__tests__/commands/Generate.test.ts`

**Comment by millsp** [comment]
Approved ✅

**Pull request merged**